scheduler/README.md

```diff
@@ -0,0 +1,192 @@
+desired allocations on feasibile nodes. Consult [Nomad documentation][0] for
+more details.
+
+The diagram below illustrates this process for the service and system schedulers
+in more detail:
+
+```
+                                 ┌──────────────┐        ┌───────────┐                                               ┌─────────────┐      ┌──────────┐
+                                 │   cluster    │        │feasibility│                          ┌─────────────┐      │    score    │      │   plan   │
+    Service and batch jobs:      │reconciliation│───────▶│   check   │─────────────────────────▶│     fit     │─────▶│ assignment  │─────▶│submission│
+                                 └──────────────┘        └───────────┘                          └─────────────┘      └─────────────┘      └──────────┘
+
+─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─
+
+                                                         ┌───────────┐     ┌──────────────┐                                               ┌──────────┐
+ System and sysbatch jobs:                               │feasibility│     │     node     │     ┌─────────────┐                           │   plan   │
+                                                         │   check   │────▶│reconciliation│────▶│     fit     │──────────────────────────▶│submission│
+                                                         └───────────┘     └──────────────┘     └─────────────┘                           └──────────┘
+```
+
+## Cluster reconciliation
+
+The first step for the service and bach job scheduler is called
+"reconcilliation," and its logic lies in `scheduler/reconcile.go` file. The
+`allocReconciler` object has one public method: `Compute`, which takes no
+arguments and returns `reconcileResults` object. This results object tells the
+scheduler about desired deployment to be updated or created, which allocations
+to place, which should be updated destructively or in-place, which should be
+stopped, and which are disconnected or are reconnecting. The reconciler works
+in terms of "buckets," that is, it processes allocations by putting them into
+different sets, and that's how its whole logic is implemented.
+
+The following vocabulary is used by the reconciler:
+
+- "tainted node:" a node is considered "tainted" if allocations must be migrated
+off of it. These are nodes that are draining or have been drained, but also
+nodes that are disconnected and should be used to calculated reconnect timeout.
+The cluster reconciler commonly also refers to "untainted" allocations, i.e.,
+those that do not require migration and are not on disconnected or reconnecting
+nodes.
+
+- "paused deployment:" a deployment is paused when it has an explicit `paused`
+status, but also when it's pending or initializing.
+
+- "migrating allocations:" allocations that are on nodes that are draining.
```

Comment:
These 6 groups are the current "buckets", right? Maybe we should call that out for each of these lines to make that clear?

Reply:
4208bfe